freqtrade/freqtradebot.py

```diff
@@ -657,6 +665,40 @@ def handle_trade(self, trade: Trade) -> bool:
         logger.info('Found no sell signals for whitelisted currencies. Trying again..')
         return False
 
+    def handle_stoploss_on_exchage(self, trade: Trade) -> bool:
+        # Check uf trade is fulfulled in which case the stoploss
+        # on exchange should be added immediately if stoploss on exchnage
+        # is on
+        result = False
+        if trade.is_open and trade.open_order_id is None and trade.stoploss_order_id is None:
+            if self.edge:
+                stoploss = self.edge.stoploss(pair=trade.pair)
+            else:
+                stoploss = self.strategy.stoploss
+
+            stop_price = trade.open_rate * (1 + stoploss)
+
+            # limit price should be less than stop price.
+            # 0.98 is arbitrary here.
+            limit_price = stop_price * 0.98
+
+            stoploss_order_id = self.exchange.stoploss_limit(
+                pair=trade.pair, amount=trade.amount, stop_price=stop_price, rate=limit_price
+            )['id']
+            trade.stoploss_order_id = str(stoploss_order_id)
+
+        # Or Check if there is a stoploss on exchnage and it is hit
+        elif trade.stoploss_order_id:
+            logger.debug('Handling stoploss on exchange %s ...', trade)
+            order = self.exchange.get_order(trade.stoploss_order_id, trade.pair)
+            if order['status'] == 'closed':
```

Comment:
i think you might also need to check for `status == "cancelled"` here - but i'm not 100% certain yet if that case would ever reach that point ...

Reply:
if stoploss is cancelled that means sell happened before. so we don't reach to this point ...